app/code/core/Mage/Core/Model/Config.php

```diff
@@ -736,17 +801,16 @@ protected function _getDeclaredModuleFiles()
             $name = explode(DIRECTORY_SEPARATOR, $v);
             $name = substr($name[count($name) - 1], 0, -4);
 
-            if ($name == 'Mage_All') {
-                $collectModuleFiles['base'][] = $v;
-            } elseif (substr($name, 0, 5) == 'Mage_') {
-                $collectModuleFiles['mage'][] = $v;
+            if (in_array($name, self::MAGE_MODULES, true)) {
```

Comment:
If `MAGE_MODULES` is define with array keys as values: `'Mage_Core' => 0, 'Mage_Eav' => 1,`, then the more performant `array_key_exists` can be used.

Reply:
You are absolutely right, thanks for the tip! Implemented directly. 👍